GomokuClient/packages/gomoku-core/src/hooks/useTiles.ts

```diff
@@ -1,22 +1,16 @@
 import { useCallback, useEffect, useRef, useState } from "react";
 import { findWinner, Winner } from "@/utils";
 import { SwaggerTypes } from "@/api";
+import { genParser } from "@/utils/getParser";
 
 //TODO: this should come from server side generated types
 export type TileColor = "black" | "white" | null;
 
-export const useTiles = (
-  boardSize: SwaggerTypes.CreateGameResponse["boardSize"],
-) => {
-  const [tiles, setTiles] = useState<TileColor[][]>(
-    Array(boardSize)
-      .fill(null)
-      .map(() => Array(boardSize).fill(null)),
+export const useTiles = (gameHistory: SwaggerTypes.GetGameHistoryResponse) => {
+  const [tiles, setTiles] = useState<TileColor[][]>(genParser(gameHistory.gen));
```

Comment:
For default state if gameHistory doesn't exist we should assign 
```js
  x: 0,
  y: 0,
```

Reply:
What line exactly this comment is about? lastTile or tiles? And how gameHistory can not exist? As soon as we create game, history exists, just some properties are null. lastTile will be null initially, I don't there is any issue with part of code

Reply:
I think no need for this specific case. Gen will be always in history, but if client were not able to fetch gameHistory, this probably should be handled at router level and if history is not received, show different view, that something went wrong

Reply:
This is even handled already:

![image](https://github.com/user-attachments/assets/31aea1ec-fb9b-44eb-b6e8-c1ec6d938870)


So, I guess everything should go smooth without that state fallback